### ISSUE TITLE
Crash on exit: pointer being freed was not allocated

### ISSUE THREAD

**Issue by ryandesign** [svn-all-fast-export/svn2git#5]
svn-all-fast-export (the latest version of the code in this repository as of 2015-12-01), installed using the svn2git port I recently added to MacPorts, crashes after doing a conversion, on OS X 10.11 El Capitan, built using Qt 5.5.1, Subversion 1.9.3 and APR 1.5.2 also installed by MacPorts.

The terminal output ends with:

```
svn-all-fast-export(14669,0x7fff71d09000) malloc: *** error for object 0x7fc2ca80d400: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```

The crash log OS X writes says:

```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called
*** error for object 0x7fc2ca80d400: pointer being freed was not allocated


Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib              0x00007fff8a619002 __pthread_kill + 10
1   libsystem_pthread.dylib             0x00007fff856ff5c5 pthread_kill + 90
2   libsystem_c.dylib                   0x00007fff8efa46e7 abort + 129
3   libsystem_malloc.dylib              0x00007fff93ab0070 free + 425
4   libapr-1.0.dylib                    0x0000000109b071b3 apr_allocator_destroy + 27
5   libapr-1.0.dylib                    0x0000000109b07761 apr_pool_terminate + 36
6   libsystem_c.dylib                   0x00007fff8efa546b __cxa_finalize_ranges + 345
7   libsystem_c.dylib                   0x00007fff8efa576f exit + 55
8   svn-all-fast-export                 0x0000000109a4b89b start + 59
```

I don't know if this is relevant, but [this Subversion mailing list post](http://svn.haxx.se/users/archive-2009-05/0968.shtml) says crashes in `apr_pool_terminate` could be due to failing to clear an error with `svn_error_clear`.

**Comment by gkello** [comment]
I have similar error in linux machine, Debian testing (stretch), svn-all-fast-export compiled with svn 1.9.3 libraries. "Segmentation fault" happens when program exists. I have been able to locate one of the commits which does cause the segfault at the end (i.e. running svn-all-fast-export with --resume-from 333 --max-rev 333 ends with segfault) The commit is ceration of branch with --parents falg, i.e. 
`svn copy http://server/project1/trunk http://server/project1/versions/project1_1.2.3/project1 --parents`
where versions folder does exists but project1_1.2.3 is automatically created.
AFAIK the same rules file works correctly when executed in debian 8.2 jessie (svn 1.8 libraries, all installed from Debian repositories)

I tried to create a small test repo that would show the error, but seems like the error is caused by some other factor that svn copy --parents. AFAIK my repo has more revisions that are causing the error, I could try to locate them if it helps You.

Can't give the repo for testing, nor publish the core dump file, I'm afraid... But would be happy to assist debugging if someone teaches what to do
